code/modules/mob/living/basic/minebots/minebot_abilities.dm

```diff
@@ -49,3 +49,118 @@
 
 	to_chat(owner, span_notice("You toggle your meson vision [(owner.sight & SEE_TURFS) ? "on" : "off"]."))
 
+/datum/action/cooldown/mob_cooldown/missile_launcher
+	name = "Launch Missile"
+	button_icon = 'icons/obj/weapons/guns/projectiles.dmi'
+	button_icon_state = "84mm-heap"
+	background_icon_state = "bg_default"
+	overlay_icon_state = "bg_default_border"
+	desc = "Launch a missile towards the target!"
+	cooldown_time = 10 SECONDS
+	shared_cooldown = NONE
+	melee_cooldown_time = 0 SECONDS
+	///how long before we launch said missile
+	var/wind_up_timer = 1 SECONDS
+
+/datum/action/cooldown/mob_cooldown/missile_launcher/Activate(atom/target)
+	var/turf/target_turf = get_turf(target)
+	if(isnull(target_turf) || !can_see(owner, target_turf, 7))
+		return FALSE
+	owner.Shake(duration = wind_up_timer)
+	addtimer(CALLBACK(src, PROC_REF(launch_missile), target_turf), wind_up_timer)
+	StartCooldown()
+	return TRUE
+
+/datum/action/cooldown/mob_cooldown/missile_launcher/proc/launch_missile(turf/target_turf)
+	new /obj/effect/temp_visual/mook_dust(get_turf(owner))
+	var/obj/effect/temp_visual/rising_rocket/new_rocket = new(get_turf(owner))
+	addtimer(CALLBACK(src, PROC_REF(drop_missile), target_turf), new_rocket.duration)
+
+/datum/action/cooldown/mob_cooldown/missile_launcher/proc/drop_missile(turf/target_turf)
+	new /obj/effect/temp_visual/falling_rocket(target_turf)
+	var/obj/effect/temp_visual/falling_shadow = new /obj/effect/temp_visual/shadow_telegraph(target_turf)
+	animate(falling_shadow, transform = matrix().Scale(0.1, 0.1), time = falling_shadow.duration)
+
+/datum/action/cooldown/mob_cooldown/drop_landmine
+	name = "Landmine"
+	button_icon = 'icons/obj/weapons/grenade.dmi'
+	button_icon_state = "landmine"
+	background_icon_state = "bg_default"
+	overlay_icon_state = "bg_default_border"
+	desc = "Drop a landmine!"
+	cooldown_time = 10 SECONDS
+	shared_cooldown = NONE
+	melee_cooldown_time = 0 SECONDS
+	click_to_activate = FALSE
+
+/datum/action/cooldown/mob_cooldown/drop_landmine/Activate(atom/target)
+	var/turf/my_turf = get_turf(owner)
+	if(isopenspaceturf(my_turf))
```

Comment:
I think you'd want `isgroundlessturf`?
```suggestion
	if(isgroundlessturf(my_turf))
```
Unless for some reason you want to allow them to lay mines on space but not openspace